Undefined symbols for architecture i386 even though it's omitted from VALID_ARCHS

* [✓] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?
I'm creating a private pod that depends on a static library **Sinch**, `s.dependency 'SinchRTC', '~> 3.12'`. And linting it by `pod lib lint --private --verbose --allow-warnings`

## What did you expect to happen?
I expected the project lints successfully.

## What happened instead?
When linting the private pod this error is raised:
```
    Undefined symbols for architecture i386:
      "_SINCallKey", referenced from:
          FalconVOIP.SINCallKitProvider.(callDidEnd in _BCA27084B1F5974A347EFFCBCAE637A6)(Foundation.Notification) -> () in FalconVOIP(SINCallKitProvider.o)
      "_SINCallDidEstablishNotification", referenced from:
          function signature specialization <Arg[0] = Owned To Guaranteed> of FalconVOIP.SINCallKitProvider.(in _BCA27084B1F5974A347EFFCBCAE637A6).init(with: __ObjC.SINClient) -> FalconVOIP.SINCallKitProvider in FalconVOIP(SINCallKitProvider.o)
      "_SINCallDidEndNotification", referenced from:
          function signature specialization <Arg[0] = Owned To Guaranteed> of FalconVOIP.SINCallKitProvider.(in _BCA27084B1F5974A347EFFCBCAE637A6).init(with: __ObjC.SINClient) -> FalconVOIP.SINCallKitProvider in FalconVOIP(SINCallKitProvider.o)
    ld: symbol(s) not found for architecture i386
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

I tried to skip the **i386** architecture by `s.pod_target_xcconfig = { 'VALID_ARCHS' => 'x86_64 armv7 arm64' }` but the same error appears.

I tried with different architectures, the only combination that skips the error is `s.pod_target_xcconfig = { 'VALID_ARCHS' => 'x86_64 arm64' }` but I want to support **armv7**.

I wanted to make sure what architectures **Sinch** supports, and I found it supports **armv7**.
```
> lipo -info Sinch.framework/Sinch
Architectures in the fat file: Sinch are: armv7 x86_64 arm64
```

**What should I do, I want to support **armv7** in the pod?**

## CocoaPods Environment

```
   CocoaPods : 1.5.3
        Ruby : ruby 2.3.7p456 (2018-03-28 revision 63024) [universal.x86_64-darwin17]
    RubyGems : 2.5.2.3
        Host : Mac OS X 10.13.6 (17G65)
       Xcode : 9.4 (9F1027a)
         Git : git version 2.14.2
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib
Repositories : bitbucket-pod-falcon-specs - git@bitbucket.org:technivance/pod-falcon-specs.git @ 1e5ceb71870616630cb49a1eb936b2d98943e449
               master - https://github.com/CocoaPods/Specs.git @ de8be8a44096debe0af8e9139b56cd6e4266258d
               Pod-Falcon-Specs - git@bitbucket.org:technivance/pod-falcon-specs.git @ 1e5ceb71870616630cb49a1eb936b2d98943e449
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.2
cocoapods-packager    : 1.5.0
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.0
cocoapods-try         : 1.1.0
```

## Project that demonstrates the issue
https://github.com/Basim-Alamuddin/Sinch-Pod-Lint-Demo.git
Simply clone and lint the pod:
```
git clone https://github.com/Basim-Alamuddin/Sinch-Pod-Lint-Demo.git
pod lib lint --private --verbose --allow-warnings
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -50,6 +50,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
   [Paul Beusterien](https://github.com/paulb777)
   [#8156](https://github.com/CocoaPods/CocoaPods/issues/8156)
 
+* Fix linting when armv7 is included but i386 isn't.  
+  [Paul Beusterien](https://github.com/paulb777)
+  [#8129](https://github.com/CocoaPods/CocoaPods/issues/8129)
+
 * Provide an installation option to disable usage of input/output paths.  
   [Dimitris Koutsogiorgas](https://github.com/dnkoutso)
   [#8073](https://github.com/CocoaPods/CocoaPods/issues/8073)

--- a/lib/cocoapods/validator.rb
+++ b/lib/cocoapods/validator.rb
@@ -921,6 +921,14 @@ module Pod
       when :ios
         command += %w(CODE_SIGN_IDENTITY=- -sdk iphonesimulator)
         command += Fourflusher::SimControl.new.destination(:oldest, 'iOS', deployment_target)
+        xcconfig = consumer.pod_target_xcconfig
+        if xcconfig
+          archs = xcconfig['VALID_ARCHS']
+          if archs && (archs.include? 'armv7') && !(archs.include? 'i386') && (archs.include? 'x86_64')
+            # Prevent Xcodebuild from testing the non-existent i386 simulator if armv7 is specified without i386
+            command += %w(ARCHS=x86_64)
+          end
+        end
       when :watchos
         command += %w(CODE_SIGN_IDENTITY=- -sdk watchsimulator)
         command += Fourflusher::SimControl.new.destination(:oldest, 'watchOS', deployment_target)

